base/tuple.jl

```diff
@@ -148,19 +154,28 @@ function map(f, t::Tuple, s::Tuple)
 end
 function map(f, t::Any16, s::Any16)
     n = length(t)
-    A = Array{Any}(n)
+    A = Array{Any,1}(n)
     for i = 1:n
         A[i] = f(t[i], s[i])
     end
-    (A...,)
+    (A...)
 end
 # n argument function
-heads() = ()
-heads(t::Tuple, ts::Tuple...) = (t[1], heads(ts...)...)
-tails() = ()
-tails(t::Tuple, ts::Tuple...) = (tail(t), tails(ts...)...)
-map(f, ::Tuple{}, ts::Tuple...) = ()
-map(f, t1::Tuple, t2::Tuple, ts::Tuple...) = (f(heads(t1, t2, ts...)...), map(f, tails(t1, t2, ts...)...)...)
+heads(ts::Tuple...) = map(t -> t[1], ts)
+tails(ts::Tuple...) = map(tail, ts)
+map(f, ::Tuple{}...) = ()
```

Comment:
I missed the beta tagging. Should this had to be merged before? If so, is there a way to checked if this brakes packages? Worse case, I can just drop the last commit, but wanted to now if it could be included.